File: capture_the_flag.js
var htmlparser = require("htmlparser");

var handler = new htmlparser.DefaultHandler(function (error, dom) {
  if (error) {
    console.log(error);
  }
  if(dom)
  {
    // console.log("I am a thing!!!");
    // console.log(dom[2]["children"][2]["children"]);
    for(let child of dom[2]["children"][2]["children"]){
      console.log(child["children"]);
    }
  }
});

var parser = new htmlparser.Parser(handler);

fetch('https://tns4lpgmziiypnxxzel5ss5nyu0nftol.lambda-url.us-east-1.on.aws/challenge')
  .then(response => response.text())
  .then(html => {
    // Step 2: Parse the HTML (if not already in a browser context)
    const doc = parser.parseComplete(html);

    // Step 3: Search for the key information
    console.log(doc);
  });

File: capture_the_flag.py
import requests
from bs4 import BeautifulSoup

url = "https://tns4lpgmziiypnxxzel5ss5nyu0nftol.lambda-url.us-east-1.on.aws/challenge"
response = requests.get(url)
html = response.content

soup = BeautifulSoup(html, 'html.parser')

pattern_elements = soup.find_all('code', class_="ramp")

# print(pattern_elements)

new_url = ""

for child in pattern_elements:
    div = child.find('div', class_="ramp")
    if div:
        span = div.find('span', class_="ramp")
        if span:
            i = span.find('i', class_="ramp char")
            if i:
                new_url += i.get("value")


print(new_url)